guides/plugins/plugins/storefront/add-scss-variables.md

```diff
@@ -0,0 +1,212 @@
+
+{% code title="<plugin root>/src/Resources/app/storefront/src/scss/base.scss" %}
+```scss
+// The value will be overwritten by the subscriber when the plugin is installed and activated
+$sass-plugin-header-bg-color: #ffcc00 !default;
+
+.header-main {
+    background-color: $sass-plugin-header-bg-color;
+}
+```
+{% endcode %}
+
+## Theme variables subscriber
+
+You can add a new subscriber according to the [Listening to events](../plugin-fundamentals/listening-to-events.md) guide.
+In this example we name the subscriber `ThemeVariableSubscriber.php`. The subscriber listens to the `ThemeCompilerEnrichScssVariablesEvent`.
+
+{% tabs %}
+{% tab title="<plugin root>/src/Subscriber/ThemeVariablesSubscriber.php" %}
+```php
+<?php declare(strict_types=1);
+
+namespace Swag\BasicExample\Subscriber;
+
+use Shopware\Storefront\Event\ThemeCompilerEnrichScssVariablesEvent;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class ThemeVariablesSubscriber implements EventSubscriberInterface
+{
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            ThemeCompilerEnrichScssVariablesEvent::class => 'onAddVariables'
+        ];
+    }
+
+    public function onAddVariables(ThemeCompilerEnrichScssVariablesEvent $event)
+    {
+        // Will render: $sass-plugin-header-bg-color: "#59ccff";
+        $event->addVariable('sass-plugin-header-bg-color', '#59ccff');
+    }
+}
+```
+{% endtab %}
+{% tab title="<plugin root>/src/Resources/config/services.xml" %}
+```xml
+<?xml version="1.0" ?>
+
+<container xmlns="http://symfony.com/schema/dic/services"
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+    
+    <services>
+        <service id="Swag\BasicExample\Subscriber\ThemeVariablesSubscriber">
+            <tag name="kernel.event_subscriber"/>
+        </service>
+    </services>
+</container>
+```
+{% endtab %}
+{% endtabs %}
+
+The `ThemeCompilerEnrichScssVariablesEvent` provides the `addVariable()` method which takes the following parameters:
+
+* `$name:` (string): The name of the SCSS variable. The passed string will be exactly in your SCSS so please be careful with special characters. We recommend using kebab-case here. The variable prefix `$` will be added automatically. We also recommend prefixing your variable name with your plugin's or company's name to prevent naming conflicts.
+* `$value:` (string): The value which should be assigned to the SCSS variable.
+* `$sanitize` (bool - optional): Optional parameter to remove special characters from the variables value. The parameter will also add quotes around the variables value. In most cases quotes are not needed e.g. for color hex values. But there may be situations where you want to pass individual strings to your SCSS variable.
+
+{% hint style="info" %}
```

Comment:
```suggestion
{% hint style="warning" %}
```

... or is this too drastic?

Reply:
Depends on the situation. I think it is more of a warning than an info as it will impact lots and thus can break lots. Vote for warning